### ISSUE TITLE
Clear button does not trigger filterChange

### ISSUE THREAD

**Issue by zhao125787935** [telerik/kendo-angular#424]
Hi Kendo team,

I find an combobox issue in my project, then I turn to official demo [Filter](http://www.telerik.com/kendo-angular-ui/components/dropdowns/combobox/filtering/) , I find same issue, date is not updated after I clear the filtered value.

Steps to reproduce:
1. Open combobox , you will see there are three items: 
![image](https://cloud.githubusercontent.com/assets/15974161/24387065/8ea14818-13a5-11e7-95cd-fc274ef70d57.png)
2. Filter item "Small"
![image](https://cloud.githubusercontent.com/assets/15974161/24387086/a4ce3452-13a5-11e7-99d5-0e02a78b176f.png)
3. Then click "clearButton". open combobox again, you will see items only have one "Small"
![image](https://cloud.githubusercontent.com/assets/15974161/24387099/c158600c-13a5-11e7-819a-f0954f5929cf.png)

Expected result:
Combobox should list all the items, because right now no item is selected or filtered.

Would you please help with this issue ?

Thanks

**Comment by valchev** [comment]
@zhao125787935 thank you for bringing this to our attention.

Actually the issue occurs because the clear button does not trigger `filterChange` event. You can workaround this by resetting the data at `valueChange` callback.

```
    handleValueChange(value) {
        if (!value) {
            this.data = this.source.slice();
        }
    }
```

[plnkr](http://plnkr.co/edit/QauedlSvDL0vd20L9zVI?p=preview)

We will fix the issue in short time by triggering the `filterChange` event when the clear button is clicked.

**Comment by tapopov** [comment]
Should be fixed in v0.39.9.
